sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/ExecutorResourceMapper.java

```diff
@@ -0,0 +1,136 @@
+package com.mesosphere.sdk.offer.evaluate;
+
+import com.google.protobuf.TextFormat;
+import com.mesosphere.sdk.offer.ResourceCollectionUtils;
+import com.mesosphere.sdk.specification.PodSpec;
+import com.mesosphere.sdk.specification.VolumeSpec;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.mesos.Protos;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Created by gabriel on 5/31/17.
+ */
+public class ExecutorResourceMapper {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final PodSpec podSpec;
+    private final Protos.ExecutorInfo executorInfo;
+    private final Collection<VolumeSpec> volumeSpecs;
+    private final List<Protos.Resource> resources;
+    private final List<Protos.Resource> orphanedResources = new ArrayList<>();
+    private final List<OfferEvaluationStage> evaluationStages;
+
+    static class VolumeLabels {
+        private final VolumeSpec volumeSpec;
+        private final String resourceId;
+        private final String persistenceId;
+
+        private VolumeLabels(VolumeSpec volumeSpec, String resourceId, String persistenceId) {
+            this.volumeSpec = volumeSpec;
+            this.resourceId = resourceId;
+            this.persistenceId = persistenceId;
+        }
+
+        @Override
+        public String toString() {
+            return ToStringBuilder.reflectionToString(this);
+        }
+    }
+
+    public ExecutorResourceMapper(PodSpec podSpec, Protos.ExecutorInfo executorInfo) {
+        this.podSpec = podSpec;
+        this.executorInfo = executorInfo;
+        this.volumeSpecs = podSpec.getVolumes();
+        this.resources = executorInfo.getResourcesList();
+        this.evaluationStages = getEvaluationStagesInternal();
+    }
+
+    public List<Protos.Resource> getOrphanedResources() {
+        return orphanedResources;
+    }
+
+    public Collection<? extends OfferEvaluationStage> getEvaluationStages() {
+        return evaluationStages;
+    }
+
+    private List<OfferEvaluationStage> getEvaluationStagesInternal() {
+        List<VolumeSpec> remainingVolumeSpecs = new ArrayList<>();
+        remainingVolumeSpecs.addAll(volumeSpecs);
+
+        List<VolumeLabels> matchingVolumes = new ArrayList<>();
+        for (Protos.Resource resource : resources) {
```

Comment:
Executor's only have Volumes so the logic's a bit simpler over here.  I'll factor out some of the logic after I get integration tests passing.